File: backend/backend/backend/src/main/java/com/example/expensetracker/repository/ExpenseRepository.java
package com.example.expensetracker.repository;



import com.example.expensetracker.model.Expense;
import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.List;

public interface ExpenseRepository extends MongoRepository<Expense, String> {

    List<Expense> findByUserId(String userId);

    List<Expense> findByUserIdAndCategory(String userId, String category);
}

File: frontend/app/src/main/java/com/lokeshdawkar/expensetracker/repository/ExpenseRepository.kt
package com.lokeshdawkar.expensetracker.repository

import com.lokeshdawkar.expensetracker.model.Expense
import com.lokeshdawkar.expensetracker.remote.RetrofitInstance


class ExpenseRepository {
    private val api = RetrofitInstance.api

    suspend fun getExpenses(userId: String, category: String?, date: String?) =
        api.getExpenses(userId, category, date)

    suspend fun addExpense(expense: Expense) =
        api.addExpense(expense)

    suspend fun updateExpense(id: String, expense: Expense) =
        api.updateExpense(id, expense)

    suspend fun deleteExpense(id: String) =
        api.deleteExpense(id)
}
